frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts

```diff
@@ -0,0 +1,422 @@
+    if (ckEditorInstance) {
+      setTimeout(() => ckEditorInstance.editing.view.focus(), 10);
+    }
+  }
+
+  quote(event:Event) {
+    event.preventDefault();
+    const target = event.currentTarget as HTMLElement;
+    const userName = target.dataset.userNameParam as string;
+    const content = target.dataset.contentParam as string;
+
+    this.openEditorWithInitialData(this.quotedText(content, userName));
+  }
+
+  private quotedText(rawComment:string, userName:string) {
+    const quoted = rawComment.split('\n')
+      .map((line:string) => `\n> ${line}`)
+      .join('');
+
+    return `${userName}\n${quoted}`;
+  }
+
+  openEditorWithInitialData(quotedText:string) {
+    this.showForm();
+    const ckEditorInstance = this.getCkEditorInstance();
+    if (ckEditorInstance && ckEditorInstance.getData({ trim: false }).length === 0) {
+      ckEditorInstance.setData(quotedText);
+    }
+  }
+
+  clearEditor() {
+    this.getCkEditorInstance()?.setData('');
+  }
+
+  hideEditorIfEmpty() {
+    const ckEditorInstance = this.getCkEditorInstance();
+    if (ckEditorInstance && ckEditorInstance.getData({ trim: false }).length === 0) {
+      this.buttonRowTarget.classList.remove('d-none');
+      this.formRowTarget.classList.add('d-none');
+
+      if (this.journalsContainerTarget) {
+        this.journalsContainerTarget.style.marginBottom = '';
+        this.journalsContainerTarget.classList.add('with-initial-input-compensation');
+        this.journalsContainerTarget.classList.remove('with-input-compensation');
+      }
+    }
+  }
+
+  async onSubmit(event:Event | null = null) {
+    event?.preventDefault();
+    const ckEditorInstance = this.getCkEditorInstance();
+    const data = ckEditorInstance ? ckEditorInstance.getData({ trim: false }) : '';
+
+    const formData = new FormData(this.formTarget);
+    formData.append('last_update_timestamp', this.lastUpdateTimestamp);
+    formData.append('filter', this.filterValue);
+    formData.append('journal[notes]', data);
+
+    const response = await this.fetchWithCSRF(this.formTarget.action, 'POST', formData);
+
+    if (response.ok) {
+      this.setLastUpdateTimestamp();
+      const text = await response.text();
+      Turbo.renderStreamMessage(text);
+
+      if (this.journalsContainerTarget) {
+        this.clearEditor();
+        if (this.isMobile()) {
+          this.hideEditorIfEmpty();
+        } else {
+          this.focusEditor();
+        }
+        if (this.journalsContainerTarget) {
+          this.journalsContainerTarget.style.marginBottom = '';
+          this.journalsContainerTarget.classList.add('with-input-compensation');
+        }
+        setTimeout(() => {
+          this.scrollJournalContainer(
+            this.journalsContainerTarget,
+            this.sortingValue === 'asc',
+          );
+          if (this.isMobile()) {
+            this.scrollInputContainerIntoView(300);
+          }
+        }, 10);
+      }
+    }
+  }
```

Comment:
This block is very hard to read, could you extract it into a method to remove the nesting of statements?